src/pages/index.js

```diff
@@ -63,9 +96,25 @@ profileEditButton.addEventListener("click", () => {
 });
 
 function handleProfileFormSubmit({ title, description }) {
-  userInfo.setUserInfo({ name: title, job: description });
-  popupProfileEdit.close();
-  validateProfile.toggleButtonState();
+  const saveButton = document.querySelector("#profile-submit-button");
+  const originalText = saveButton.textContent;
+
+  saveButton.textContent = "Saving...";
```

Comment:
Needs correction: the logic for rendering loading gets duplicated. You should create a function or method and use it.

Better to make the code for rendering loading more flexible

You can manage loading via `PopupWithForm`:

```js
constructor(popupSelector, { loadingButtonText, ... }) { 
  ...
  this._buttonText = this._submitButton.textContent;
  this._loadingButtonText = loadingButtonText;
}

showLoading() {
  this._submitButton.textContent = this._loadingButtonText
}

hideLoading() {
  this._submitButton.textContent = this._buttonText
}
```

```js
// index.js
profileEditPopup.showLoading()
...
.finally(() => {
 profileEditPopup.hideLoading()
})
```